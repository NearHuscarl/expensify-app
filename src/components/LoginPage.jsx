import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const LoginPage = (props) => (
	<div>
		<input type="text" placeholder="abc@gmail.com" autoFocus />
		<input type="password" placeholder="your password" />
		<button onClick={props.startLogin}>Login</button>
	</div>
);

const mapDispatchToProps = (dispatch) => ({
	startLogin: () => dispatch(startLogin()),
});

export default connect(
	undefined,
	mapDispatchToProps,
)(LoginPage);
