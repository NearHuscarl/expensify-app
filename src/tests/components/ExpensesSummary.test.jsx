import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render ExpenseSummary correctly with 1 expense', () => {
	const wrapper = shallow(
		<ExpensesSummary expensesCount={1} expensesTotal={9434} />,
	);
	expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseSummary correctly with multiple expenses', () => {
	const wrapper = shallow(
		<ExpensesSummary expensesCount={2} expensesTotal={9434} />,
	);
	expect(wrapper).toMatchSnapshot();
});

test('should not render ExpenseSummary with 0 hidden expense', () => {
	const wrapper = shallow(
		<ExpensesSummary hiddenExpensesCount={0} expensesTotal={0} />,
	);
	expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseSummary correctly with 1 hidden expense', () => {
	const wrapper = shallow(
		<ExpensesSummary hiddenExpensesCount={1} expensesTotal={9434} />,
	);
	expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseSummary correctly with multiple hidden expenses', () => {
	const wrapper = shallow(
		<ExpensesSummary hiddenExpensesCount={20} expensesTotal={9434} />,
	);
	expect(wrapper).toMatchSnapshot();
});
