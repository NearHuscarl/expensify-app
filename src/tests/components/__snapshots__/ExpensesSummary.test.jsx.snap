// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not render ExpenseSummary with 0 hidden expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span />
       
      expenses
       totalling
       
      <span>
        $0.00
      </span>
    </h1>
    <div
      className="page-header__action"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should render ExpenseSummary correctly with 1 expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       totalling
       
      <span>
        $94.34
      </span>
    </h1>
    <p
      className="page-header__subtitle"
    >
       hidden 
      expenses
    </p>
    <div
      className="page-header__action"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should render ExpenseSummary correctly with 1 hidden expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span />
       
      expenses
       totalling
       
      <span>
        $94.34
      </span>
    </h1>
    <p
      className="page-header__subtitle"
    >
      1
       hidden 
      expense
    </p>
    <div
      className="page-header__action"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should render ExpenseSummary correctly with multiple expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        2
      </span>
       
      expenses
       totalling
       
      <span>
        $94.34
      </span>
    </h1>
    <p
      className="page-header__subtitle"
    >
       hidden 
      expenses
    </p>
    <div
      className="page-header__action"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should render ExpenseSummary correctly with multiple hidden expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span />
       
      expenses
       totalling
       
      <span>
        $94.34
      </span>
    </h1>
    <p
      className="page-header__subtitle"
    >
      20
       hidden 
      expenses
    </p>
    <div
      className="page-header__action"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
